RecoTracker/LSTCore/src/alpaka/PT2.h

```diff
@@ -0,0 +1,343 @@
+      alpaka::wait(queue);
+    }
+
+    inline PT2s const* data() const { return &data_; }
+    inline void setData(PT2sBuffer& buf) { data_.setData(buf); }
+  };
+
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE void addPT2ToMemory(lst::MiniDoublets const& mdsInGPU,
+                                                              lst::Segments const& segmentsInGPU,
+                                                              lst::PT2s& PT2sInGPU,
+                                                              unsigned int pixelSegmentIndex,
+                                                              unsigned int segmentIndex,
+                                                              float pixelRadius,
+                                                              float centerX,
+                                                              float centerY,
+                                                              float rzChiSquared,
+                                                              unsigned int PT2Index,
+                                                              float pt,
+                                                              float eta,
+                                                              float phi,
+                                                              float eta_pix,
+                                                              float phi_pix) {
+    PT2sInGPU.pixelSegmentIndices[PT2Index] = pixelSegmentIndex;
+    PT2sInGPU.segmentIndices[PT2Index] = segmentIndex;
+    PT2sInGPU.pixelRadius[PT2Index] = __F2H(pixelRadius);
+    PT2sInGPU.pt[PT2Index] = __F2H(pt);
+    PT2sInGPU.eta[PT2Index] = __F2H(eta);
+    PT2sInGPU.phi[PT2Index] = __F2H(phi);
+    PT2sInGPU.eta_pix[PT2Index] = __F2H(eta_pix);
+    PT2sInGPU.phi_pix[PT2Index] = __F2H(phi_pix);
+    PT2sInGPU.isDup[PT2Index] = false;
+
+    PT2sInGPU.centerX[PT2Index] = __F2H(centerX);
+    PT2sInGPU.centerY[PT2Index] = __F2H(centerY);
+    PT2sInGPU.logicalLayers[Params_pT2::kLayers * PT2Index] = 0;
+    PT2sInGPU.logicalLayers[Params_pT2::kLayers * PT2Index + 1] = 0;
+    PT2sInGPU.logicalLayers[Params_pT2::kLayers * PT2Index + 2] =
+        segmentsInGPU.logicalLayers[segmentIndex * Params_LS::kLayers];
+    PT2sInGPU.logicalLayers[Params_pT2::kLayers * PT2Index + 3] =
+        segmentsInGPU.logicalLayers[segmentIndex * Params_LS::kLayers + 1];
+
+    PT2sInGPU.lowerModuleIndices[Params_pT2::kLayers * PT2Index] =
+        segmentsInGPU.innerLowerModuleIndices[pixelSegmentIndex];
+    PT2sInGPU.lowerModuleIndices[Params_pT2::kLayers * PT2Index + 1] =
+        segmentsInGPU.outerLowerModuleIndices[pixelSegmentIndex];
+    PT2sInGPU.lowerModuleIndices[Params_pT2::kLayers * PT2Index + 2] =
+        segmentsInGPU.innerLowerModuleIndices[Params_LS::kLayers * segmentIndex];
+    PT2sInGPU.lowerModuleIndices[Params_pT2::kLayers * PT2Index + 3] =
+        segmentsInGPU.outerLowerModuleIndices[Params_LS::kLayers * segmentIndex + 1];
```

Comment:
The indexing here is off. I'm not sure what it should be, but maybe something more like

```suggestion
    PT2sInGPU.lowerModuleIndices[Params_pT2::kLayers * PT2Index + 2] =
        segmentsInGPU.innerLowerModuleIndices[segmentIndex];
    PT2sInGPU.lowerModuleIndices[Params_pT2::kLayers * PT2Index + 3] =
        segmentsInGPU.outerLowerModuleIndices[segmentIndex];
```

Reply:
Here's a tip for debugging. Don't use `cuda-gdb`, since in CMSSW it's a script that currently is broken. Instead, you can use `cuda-gdb.real`, which is the real one and works fine.

Reply:
Thank you Andres, this is indeed incorrect, fixed